### DISCUSSION TITLE
Which Filament Permissions Plugin is Better for a Multi-Country CRM? (Althinect Spatie vs Shield)

### DISCUSSION THREAD

**Discussion by NichitaPaingu** [filamentphp/filament#15105] [Help]
### Package

Other

### Package Version

V.2

### How can we help you?

Hello everyone,

I’m working on a CRM application for a company operating in two countries. Each country will function as a separate company within the CRM, and the users will be associated with one of these two companies. The main requirement is that users can only access the records specific to their own company.

I’ve come across two Filament plugins for permission management:

[Althinect Spatie Roles & Permissions](https://v2.filamentphp.com/plugins/althinect-spatie-roles-permissions)
[Shield](https://v2.filamentphp.com/plugins/shield)
I’m trying to decide which one would be the best fit for this use case. Here’s what I’m looking for:

Data Segregation: Each company’s users should only see and interact with their own records.
Scalability: The system should allow adding more companies or countries in the future.
Flexibility: Support for both role-based and record-specific permissions.
Ease of Integration: It should work seamlessly with a Laravel-based Filament application.
Performance: Efficient handling of permissions for potentially large data sets.
If you’ve used either of these plugins or have insights into their pros and cons, I’d love to hear your recommendations. Which one would you choose for this scenario, and why?

Looking forward to your feedback!

**Comment by erenergul** [comment]
I prefer Althinech. Its easy to add in your requirement. And you dont need permission for users can only acces to records. You just need to add owner_id and and you can add your resource something like that  

public static function getEloquentQuery(): Builder
    {
        return parent::getEloquentQuery()
            ->where('owner_id', auth()->id());
    }
